engine/src/graphics/device.cpp

```diff
@@ -147,13 +165,23 @@ namespace gen
 		gen::logger::debug("vulkan", std::format("Found Queue Family Properties: \n{}\n", qProp.str()));
 
 		int index = 0;
-		for (const auto & queueFamily : queueFamilies)
+		for (const auto &queueFamily : queueFamilies)
 		{
 			if (queueFamily.queueFlags & vk::QueueFlagBits::eGraphics)
 			{
 				indices.graphicsFamily = index;
 			}
 
+			if (queueFamily.queueFlags & vk::QueueFlagBits::eTransfer)
+			{
+				indices.transferFamily = index;
+			}
+
+			if (device.getSurfaceSupportKHR(static_cast<u32>(index), surface)) // NOLINT(readability-implicit-bool-conversion)
+			{
+				indices.presentFamily = index;
+			}
+
```

Comment:
Nice, but it still needs a bit more work: we want to use a single queue from a single family that supports everything. At least one family that supports all (relevant) operations is guaranteed to be available by the spec, it's just not guaranteed to be the 0th one, so we need to iterate over them to find it.

Reply:
This is a tiny abstraction I use to represent the physical device + queue (family), in case it's useful:
```cpp
struct Gpu {
	vk::PhysicalDevice device{};
	vk::PhysicalDeviceProperties properties{};
	std::uint32_t queue_family{};
};
```